File: julia/hamming/hamming.jl
"calculate hamming distance"
function distance(strand_1::String, strand_2::String)
    if length(strand_1) != length(strand_2) throw(ArgumentError("Require equal length of vectors")) end
    s1 =  Int.(collect(strand_1))
    s2 = Int.(collect(strand_2))
    sum(s1 .- s2 .!= 0) 
end

File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

//Distance ...Calculate Hamming Distance
func Distance(a, b string) (int, error) {

	if len(a) != len(b) {
		return 0, errors.New("must be the same length")
	}

	sum := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			sum++
		}
	}
	return sum, nil
}
